src/datasets/builder.py

```diff
@@ -1205,7 +1213,7 @@ def _as_dataset(self, split: Union[ReadInstruction, Split] = Split.TRAIN, in_mem
         """
         cache_dir = self._fs._strip_protocol(self._output_dir)
         dataset_kwargs = ArrowReader(cache_dir, self.info).read(
-            name=self.name,
+            name=self.dataset_name,
```

Comment:
for packaged datasets - won't be able, true... I'm not sure what to do here because I also changed hash for packaged datasets with configs, see `update_hash_with_config_parameters()` above, and idk if it's correct.
`update_hash_with_config_parameters()` also returns hash of different length, so the following old cache dir of some audiofolder dataset:
```
audiofolder/<ds_name>-8f8242e21ba534e/0.0.0/6cbdd16f8688354c63b4e2a36e1585d05de285023ee6443ffd71c4182055c0fc/
```
now it is
```
<ds-name>/<config-name>/0.0.0/1bbaadc65faace94/
```

If not change hash here, it would be always the same for all audiofolder datasets (`6cbdd16f8688354c63b4e2a36e1585d05de285023ee6443ffd71c4182055c0fc`).

Reply:
Makes sense. Reloading from the cache in offline mode wasn't implemented for datasets without scripts anyway.

Reply:
Btw I think the `hash` is computed in `load.py`, e.g. in HubDatasetModuleFactoryWithoutScript.get_module

Maybe updating the hash for configured builders can be done there as well, to not complexify `DatasetBuilder.__init__`